test/remote/gateways/remote_merchant_partners_test.rb

```diff
@@ -0,0 +1,162 @@
+    assert_equal "Invalid account number", response.message
+    assert_equal "1102140001", response.error_code
+  end
+
+  def test_failed_capture
+    response = @gateway.capture(@amount, "BAD")
+    assert_failure response
+    assert_equal "Missing account number", response.message
+    assert_equal "1102010009", response.error_code
+  end
+
+  def test_successful_void
+    response = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success response
+
+    void = @gateway.void(response.authorization)
+    assert_success void
+    assert_equal "Succeeded", void.message
+  end
+
+  def test_failed_void
+    response = @gateway.void("")
+    assert_failure response
+    assert_equal "Invalid acct type", response.message
+    assert_equal "3101810001", response.error_code
+  end
+
+  def test_successful_refund
+    response = @gateway.purchase(@amount, @credit_card, @options)
+    assert_success response
+
+    refund = @gateway.refund(@amount, response.authorization)
+    assert_success refund
+    assert_equal "Succeeded", refund.message
+  end
+
+  def test_failed_refund
+    response = @gateway.refund(nil, "")
+    assert_failure response
+    assert_equal "Missing account number", response.message
+    assert_equal "1101700009", response.error_code
+  end
+
+  def test_successful_credit
+    response = @gateway.credit(@amount, @credit_card, @options)
+    assert_success response
+    assert_equal "Succeeded", response.message
+  end
+
+  def test_failed_credit
+    response = @gateway.credit(@amount, @declined_card, @options)
+    assert_failure response
+    assert_equal "Invalid account number", response.message
+    assert_equal "1102140001", response.error_code
+  end
+
+  def test_successful_verify
+    response = @gateway.verify(@credit_card, @options)
+    assert_success response
+    assert_match %r{Succeeded}, response.message
+  end
+
+  def test_failed_verify
+    response = @gateway.verify(@declined_card, @options)
+    assert_failure response
+    assert_equal "Invalid account number", response.message
+    assert_equal "1102140001", response.error_code
+  end
+
+  def test_successful_store_and_purchase
+    response = @gateway.store(@credit_card, @options)
+    assert_success response
+    assert_equal "Succeeded", response.message
+
+    purchase = @gateway.purchase(@amount, response.authorization, @options)
+    assert_success purchase
+    assert_equal "Succeeded", purchase.message
+  end
+
+  def test_successful_store_and_credit
+    response = @gateway.store(@credit_card, @options)
+    assert_success response
+    assert_equal "Succeeded", response.message
+
+    credit = @gateway.credit(@amount, response.authorization, @options)
+    assert_success credit
+    assert_equal "Succeeded", credit.message
+  end
+
+  def test_failed_store
+    response = @gateway.store(@declined_card, @options)
+    assert_failure response
+
+    # Test gateway bombs w/ live-transaction error so can't test
```

Comment:
Yeah, it throws an error about live transactions not being allowed for this one call only :(